onie-mk-demo.sh

```diff
@@ -93,6 +94,7 @@ sed -i -e "s/%%DEMO_TYPE%%/$demo_type/g" \
        -e "s/%%ONIE_IMAGE_PART_SIZE%%/$onie_image_part_size/" \
        -e "s/%%EXTRA_CMDLINE_LINUX%%/$EXTRA_CMDLINE_LINUX/" \
        -e "s@%%OUTPUT_RAW_IMAGE%%@$output_raw_image@" \
+       -e "s/%%ONIE_MACHINE%%/$machine/g" \
```

Comment:
>$machine [](http://example.com/codeflow?start=30&length=8)

You are building the image ground truth into `installer.sh`. 

Alternatively, I suggest that `/etc/sonic/sonic_version.yml` asic_type field is the ground truth. Then you don't need to change `installer.sh`.

This is the same to other bootloader such as aboot. In future, we may have similar logic in bootloader other than ONIE.